storagenode/piecestore/endpoint.go

```diff
@@ -578,17 +586,30 @@ func (endpoint *Endpoint) Retain(ctx context.Context, retainReq *pb.RetainReques
 				if err = endpoint.pieceinfo.Delete(ctx, peer.ID, pieceID); err != nil {
 					endpoint.log.Error("failed to delete piece info", zap.Error(Error.Wrap(err)))
 				}
+				numDeleted++
 			}
 		}
 		hasMorePieces = (len(pieceIDs) == limit)
 		offset += len(pieceIDs)
+		offset -= numDeleted
 		// We call Gosched() here because the GC process is expected to be long and we want to keep it at low priority,
 		// so other goroutines can continue serving requests.
 		runtime.Gosched()
 	}
 	return &pb.RetainResponse{}, nil
 }
 
+func (endpoint *Endpoint) isTrustedSatellite(ctx context.Context, peer *identity.PeerIdentity) bool {
```

Comment:
this should be in the pool implementation

Reply:
that is, a method on `trust` instead of a method on `endpoint`?

Reply:
yes

Reply:
okay changed it!